src/proto/streams/store.rs

```diff
@@ -140,12 +137,15 @@ impl Store {
 
         while i < len {
             // Get the key by index, this makes the borrow checker happy
-            let key = *self.ids.get_index(i).unwrap().1;
+            let (stream_id, index) = {
```

Comment:
Nit: I think this can be simplified to just
```rust
let (&stream_id, &index) = self.ids.get_index(i).unwrap();
```

Reply:
Does that require a more recent compiler? I worried the lifetime would cause a conflict with passing `self` into `f()`...